python/ray/autoscaler/v2/instance_manager/instance_manager.py

```diff
@@ -63,3 +66,205 @@ def get_instance_manager_state(
             The reply to the request.
         """
         pass
+
+
+class InstanceUtil:
+    """
+    A helper class to group updates and operations on an Instance object defined
+    in instance_manager.proto
+    """
+
+    @staticmethod
+    def new_instance(
+        instance_id: str,
+        instance_type: str,
+        resources: Dict[str, float],
+        request_id: str = "",
+    ) -> Instance:
+        instance = Instance()
+        instance.version = 0  # it will be populated by the underlying storage.
+        instance.instance_id = instance_id
+        instance.instance_type = instance_type
+        instance.launch_request_id = request_id
+        instance.total_resources.update(resources)
+        instance.status = Instance.QUEUED
+        InstanceUtil._record_status_transition(
```

Comment:
_record_status_transition should be a public method of `instance` instead?

Reply:
it's expected to use within `set_status` for now.